src/cluster.c

```diff
@@ -6287,6 +6288,7 @@ NULL
             return;
         }
 
+        removeAllShardChannelSubscription();
         /* Set the master. */
         clusterSetMaster(n);
```

Comment:
I was also talking with @madolson about this. `clusterSetMaster` seems the right place to put this logic to cover all scenarios. Irrespective of it being a primary or replica earlier, we could send the unsubscribe message.

Reply:
OK, if clusterSetMaster just make sure we don't unsubscribe if it was a failover because in this case all nodes are still in the same shards.

Why not do it it clusterRemoveNodeFromShard? It seems even better to me.

Reply:
`clusterRemoveNodeFromShard` just seems like a utility method. Also, the primary/replica relation needs to be intact to be able cleanup the appropriate slots.